cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DocumentStore.java

```diff
@@ -0,0 +1,138 @@
+package com.cloudant.sync.datastore;
+
+import com.cloudant.sync.datastore.encryption.KeyProvider;
+import com.cloudant.sync.datastore.encryption.NullKeyProvider;
+import com.cloudant.sync.event.EventBus;
+import com.cloudant.sync.event.Subscribe;
+import com.cloudant.sync.notifications.DatabaseClosed;
+import com.cloudant.sync.notifications.DatabaseCreated;
+import com.cloudant.sync.notifications.DatabaseDeleted;
+import com.cloudant.sync.notifications.DatabaseOpened;
+import com.cloudant.sync.query.IndexManager;
+import com.cloudant.sync.query.IndexManagerImpl;
+import com.cloudant.sync.util.Misc;
+import com.sun.javafx.scene.shape.PathUtils;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * Created by tomblench on 27/09/2016.
+ */
+
+public class DocumentStore {
+
+    public final Database database;
+    public final IndexManager query;
+    protected final String databaseName; // only used for events
+    private final File location; // needed for close
+
+    /* This map should only be accessed inside synchronized(openDatastores) blocks */
+    private static final Map<File, DocumentStore> documentStores = new HashMap<File, DocumentStore>();
+
+    private static EventBus eventBus = new EventBus();
+
+    private DocumentStore(File location, KeyProvider keyProvider) throws DatastoreException, IOException, SQLException {
+        this.location = location;
+        this.databaseName = location.getName();
+        this.database = new DatabaseImpl(location, keyProvider);
+        // this will be used to observe purge() events from database and forward them to indexmanager
+        this.database.getEventBus().register(this);
```

Comment:
@rhyshort this will be used when we implement purge which isn't part of this feature branch, we can take it out now if you don't think it's relevant here